src/app/games/checkers/common/tests/CheckersControlPlusDominationHeuristic.spec.ts

```diff
@@ -0,0 +1,128 @@
+import { CheckersPiece, CheckersStack, CheckersState } from '../CheckersState';
+import { AbstractCheckersRules, CheckersConfig } from '../AbstractCheckersRules';
+import { CheckersControlPlusDominationHeuristic } from '../CheckersControlPlusDominationHeuristic';
+import { HeuristicUtils } from 'src/app/jscaip/AI/tests/HeuristicUtils.spec';
+import { InternationalCheckersRules } from '../../international-checkers/InternationalCheckersRules';
+import { LascaRules } from '../../lasca/LascaRules';
+
+const O: CheckersStack = new CheckersStack([CheckersPiece.ZERO]);
+const X: CheckersStack = new CheckersStack([CheckersPiece.ONE]);
+const _: CheckersStack = CheckersStack.EMPTY;
+
+const rules: AbstractCheckersRules[] = [
+    InternationalCheckersRules.get(),
+    LascaRules.get(),
+];
+
+for (const rule of rules) {
+
+    describe('CheckersControlAndDominationHeuristic for ' + rule.constructor.name, () => {
+
+        const defaultConfig: MGPOptional<CheckersConfig> = rule.getDefaultRulesConfig();
+
+        let heuristic: CheckersControlPlusDominationHeuristic;
+
+        beforeEach(() => {
+            heuristic = new CheckersControlPlusDominationHeuristic(rule);
+        });
+
+        it('should not count the immobilized stacks', () => {
+            // Given two boards with the exact same stacks, one having blocked stacks
+            const immobilizedState: CheckersState = CheckersState.of([
+                [X, _, _, _, _, _, _],
+                [_, O, _, _, _, _, _],
+                [_, _, O, _, _, _, _],
+                [_, _, _, _, _, _, _],
+                [_, _, _, _, X, _, _],
+                [_, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _],
+            ], 0);
+            const mobileState: CheckersState = CheckersState.of([
+                [X, _, _, _, _, _, _],
+                [_, _, _, O, _, _, _],
+                [_, _, O, _, _, _, _],
+                [_, _, _, _, _, _, _],
+                [_, _, _, _, X, _, _],
+                [_, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _],
+            ], 0);
+
+            // When comparing them
+            // Then the one with mobile stacks should be considered better
+            HeuristicUtils.expectSecondStateToBeBetterThanFirstFor(heuristic,
+                                                                   immobilizedState,
+                                                                   MGPOptional.empty(),
+                                                                   mobileState,
+                                                                   MGPOptional.empty(),
+                                                                   Player.ONE,
+                                                                   defaultConfig);
+        });
+
+        it('should count the potential mobility as primary board value', () => {
+            // Given two boards with the same stacks, one with an unique forced capture, the other without
```

Comment:
"an unique" -> "a unique", u étant une consonne !

Reply:
wait what ?

Reply:
You liar !

Reply:
![2024-11-24_12-00](https://github.com/user-attachments/assets/94d03cc2-ea69-4b56-815a-50cc27089ae9)